guide/build-config.md

```diff
@@ -138,13 +139,17 @@ builds:
   scmUrl: https://<external git server>/<repo>.git
 ```
 
-#### environmentId v/s systemImageId
+#### environmentId v/s environmentName v/s systemImageId
 
-The build environment used for builds has a unique id and system image id.
+The build environment used for builds has a unique id, name and system image id.
 
-We allow the user to specify the build environment using either `environmentId` or the `systemImageId`. `environmentId` refers to the exact id of the build environment as stored in the PNC database. This is what we traditionally use, but it is problematic since the ids are different between PNC servers (Devel, Stage, Prod), which makes the `build-config.yaml` not very portable between PNC servers.
+We allow the user to specify the build environment using either `environmentId`, `environmentName` or the `systemImageId`. `environmentId` refers to the exact id of the build environment as stored in the PNC database. This is what we traditionally use, but it is problematic since the ids are different between PNC servers (Devel, Stage, Prod), which makes the `build-config.yaml` not very portable between PNC servers.
```

Comment:
I would add: "Moreover, an environment may become deprecated over time, thus the environmentId should be kept up-to-date, which is not optimal".

Reply:
Ack!